ROX-17766, ROX-17780: upgrade test and psps

dashrews78 opened the pull request:
## Description

The issue at hand was that recently all the gke clusters got bumped to `v1.25` of kubernetes meaning that pod security policies were no longer valid.  Since this test deploys old versions of software WITH old tooling to match it became problematic to get the old versions to deploy without pod security polices.  So I changed it from deploying with the deploy scripts to have the old versions of roxctl generate helm so I could use helm install and thus pass the flag for helm to ignore the pod security policies.


## Checklist
- [ ] Investigated and inspected CI test results
~- [ ] Unit test and regression tests added~
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Purely fixing the test.

dashrews78 committed:
X-Smart-Branch-Parent: master

porridge commented on sensor/upgrader/upgradectx/context.go:
```diff
@@ -345,3 +352,8 @@ func (c *UpgradeContext) ListCurrentObjects() ([]*unstructured.Unstructured, err
+
+// IsPodSecurityEnabled returns whether or not pod security polices are enabled for this cluster
+func (c *UpgradeContext) IsPodSecurityEnabled() bool {
+	return c.podSecurityPoliciesSupported
```
This field seems to be [set](https://github.com/stackrox/stackrox/pull/6505/files#diff-acc1578eababf29f1daa395679380b56b089fa2d0b0fe30ee316ffce9917bb1fR110) based on `OrderedBundleResourceTypes` which according to the comment will contain PSPs for all eternity, so I cannot see how this function could ever return anything else than `true`?

dashrews78 committed:
X-Smart-Squash: Squashed 16 commits:

9d94644528 try excluding PSP
c7b432d6ab forgot to continue
de20152e0e slow down and pay attention
6ea6ec26e1 preconditions too
c82d16b1f9 quick test excluding files to let CI while I work on better fix since CI takes forever
ba9affb64c untangling the web
38bdb9ee45 files
5d4df7674f about 3 different ways right now
d3829f7ef6 back to just sensor
464f092068 back to just sensor
0646f14619 try to fix the upgrader
63d62f849a negate and test
3c350cc907 try filter again before giving up and ignoring files
479ee05aa4 lint problems
c22a0a5e35 see if I can fix it by simply bumping the SW version
ea358ec241 hope

dashrews78 committed:
some testing

dashrews78 committed:
yet another round of testing

dashrews78 committed:
forgot to set the tag

dashrews78 committed:
try helm should be less fickle

dashrews78 committed:
messing with images

dashrews78 committed:
grab pass from elsewhere after initial helm install

dashrews78 committed:
debug

dashrews78 committed:
fix pw

dashrews78 committed:
try pw again

dashrews78 committed:
command is wrong

dashrews78 committed:
still fighting with pw

dashrews78 committed:
beyond the pw probs on to helm

dashrews78 committed:
images

dashrews78 committed:
try something

dashrews78 committed:
try something else

dashrews78 committed:
play with images later

dashrews78 committed:
try 3.69 again

dashrews78 committed:
back to 74

dashrews78 committed:
images

dashrews78 committed:
psp

dashrews78 committed:
clean up installs

dashrews78 committed:
switching to 74

dashrews78 committed:
some more debug

dashrews78 committed:
scale sensor work

dashrews78 commented:
/test gke-upgrade-tests

dashrews78 committed:
work on timing with lb

dashrews78 committed:
scale work

dashrews78 committed:
sensor with  helm

dashrews78 committed:
forgot to point to chart bundle

dashrews78 committed:
images for sensor

dashrews78 committed:
collecgtor

dashrews78 committed:
clean up sensor

dashrews78 committed:
clean up and hope it still works

dashrews78 committed:
slow down

dashrews78 commented:
/test gke-upgrade-tests

connorgorman commented on tests/upgrade/lib.sh:
```diff
@@ -85,20 +85,34 @@ deploy_earlier_central() {
 
     if is_CI; then
-        gsutil cp "gs://stackrox-ci/roxctl-$EARLIER_TAG" "bin/$TEST_HOST_PLATFORM/roxctl"
+        make cli
```
This can just me `make cli`

connorgorman reviewed (approved):
Looks good

dashrews78 committed:
PR comment
